pkg/nemesis/container_kill.go

```diff
@@ -0,0 +1,113 @@
+	var n int
+	var component *cluster.Component
+	var freq = time.Second * time.Duration(rand.Intn(120)+60)
+	switch g.name {
+	case "short_kill_tikv_1node":
+		n = 1
+		cmp := cluster.TiKV
+		component = &cmp
+	case "short_kill_pd_leader":
+		n = 1
+		nodes = findPDMember(nodes, true)
+		cmp := cluster.PD
+		component = &cmp
+	}
+	return containerKillNodes(nodes, n, component, freq)
+}
+
+func (g containerKillGenerator) Name() string {
+	return g.name
+}
+
+func containerKillNodes(nodes []cluster.Node, n int, component *cluster.Component, freq time.Duration) []*core.NemesisOperation {
+	var ops []*core.NemesisOperation
+	if component != nil {
+		nodes = filterComponent(nodes, *component)
+	}
+	indices := shuffleIndices(len(nodes))
+	if n > len(nodes) {
+		n = len(nodes)
+	}
+	for i := 0; i < n; i++ {
+		ops = append(ops, &core.NemesisOperation{
+			Type:        core.ContainerKill,
+			Node:        &nodes[indices[i]],
+			InvokeArgs:  nil,
+			RecoverArgs: nil,
+			// Note: Container-Kill is a instant action, Runtime means the duration of next time container kill here.
+			RunTime: freq,
+		})
+	}
+
+	return ops
+}
+
+// NewContainerKillGenerator creates a generator.
+func NewContainerKillGenerator(name string) core.NemesisGenerator {
+	return containerKillGenerator{name: name}
+}
+
+// containerKill implements Nemesis
+type containerKill struct {
+	k8sNemesisClient
+}
+
+func (k containerKill) Invoke(ctx context.Context, node *cluster.Node, args ...interface{}) error {
+	log.Printf("Creating container-kill with node %s(ns:%s)\n", node.PodName, node.Namespace)
+	containerChaos := buildContainerKillChaos(node.Namespace, node.Namespace,
+		node.PodName, string(node.Component))
+	return k.cli.ApplyPodChaos(ctx, &containerChaos)
+}
+
+func (k containerKill) Recover(ctx context.Context, node *cluster.Node, args ...interface{}) error {
+	log.Printf("Recover container-kill with node %s(ns:%s)\n", node.PodName, node.Namespace)
+	containerChaos := buildContainerKillChaos(node.Namespace, node.Namespace, node.PodName, string(node.Component))
+	return k.cli.CancelPodChaos(ctx, &containerChaos)
+}
+
+func (containerKill) Name() string {
+	return string(core.ContainerKill)
+}
+
+func buildContainerKillChaos(ns string, chaosNs string, podName string, containerName string) chaosv1alpha1.PodChaos {
+	chaos := chaosv1alpha1.ContainerKillAction
+	pods := make(map[string][]string)
+	pods[ns] = []string{podName}
+	return chaosv1alpha1.PodChaos{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      strings.Join([]string{podName, string(chaos)}, "-"),
+			Namespace: chaosNs,
+		},
+		Spec: chaosv1alpha1.PodChaosSpec{
+			Selector: chaosv1alpha1.SelectorSpec{
+				Pods: pods,
+			},
+			ContainerName: containerName,
+			Action:        chaos,
+			Mode:          chaosv1alpha1.OnePodMode,
+			Scheduler:     &chaosv1alpha1.SchedulerSpec{Cron: "@every 1000000s"}, // let it be scheduled in our control
```

Comment:
Seems this is regard a instant action?

Reply:
Sure, I think that using freq to control schedule such instant action is better.